service/cloud/gcp_streamer_internal_test.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package cloud
+
+import (
+	"testing"
+
+	"cloud.google.com/go/storage"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/optakt/flow-dps/models/dps"
+	"github.com/optakt/flow-dps/testing/mocks"
+)
+
+func TestNewGCPStreamer(t *testing.T) {
+	log := zerolog.Nop()
+	bucket := &storage.BucketHandle{}
+	limit := uint(42)
+
+	streamer := NewGCPStreamer(log, bucket, WithBufferSize(limit))
+
+	require.NotNil(t, streamer)
+	assert.Equal(t, bucket, streamer.bucket)
+	assert.Equal(t, limit, streamer.limit)
```

Comment:
> You are not checking the log here. Seems kind of useless to initialize it then.

We can check it but it's no longer equal to the given one since we called `log.With().Str("component", "gcp_streamer").Logger()` in the meantime.

What would you suggest to do with it then? Compare it against `log.With().Str("component", "gcp_streamer").Logger()`? We have to pass a logger, since it's not a pointer in this case but a struct directly that the func takes as parameter.

> Also, should check that the internal queue and buffer ar not nil.

Will do!